sim/src/generic_fss_hardware_model.cpp

```diff
@@ -0,0 +1,241 @@
+
+        // Create next queued data based on in_data
+        if ((in_data.size() == 7) && 
+            (in_data[0] == 0xDE) && (in_data[1] == 0xAD) && (in_data[2] == 0xBE) && (in_data[3] == 0xEF) &&
+            (in_data[4] == 0x01) && (in_data[5] == 0x01) && (in_data[6] == 0x02)                            ) {
+            create_generic_fss_data();
+        } else {
+            _queued_data.resize(0);
+        }
+        return out_data;
+    }
+
+    /* Custom function to prepare the Generic_fss Data */
+    void Generic_fssHardwareModel::create_generic_fss_data()
+    {
+        uint8_t four_bytes[4];
+        boost::shared_ptr<Generic_fssDataPoint> data_point = boost::dynamic_pointer_cast<Generic_fssDataPoint>(_generic_fss_dp->get_data_point());
+
+        /* Prepare data size */
+        _queued_data.resize(16, 0x00);
+
+        sim_logger->debug("Generic_fssHardwareModel::create_generic_fss_data:  Creating data, enabled=%d", _enabled);
+        if (_enabled == GENERIC_FSS_SIM_SUCCESS) {
+            bool valid = data_point->get_generic_fss_valid();
+            double alpha = data_point->get_generic_fss_alpha();
+            double beta = data_point->get_generic_fss_beta();
+            sim_logger->debug("Generic_fssHardwareModel::create_generic_fss_data:  data_point data:  valid=%s, alpha=%f, beta=%f", valid?"TRUE ":"FALSE", alpha, beta);
+
+            /* Streaming data header - 0xDEADBEEF */
+            _queued_data[0] = 0xDE;
+            _queued_data[1] = 0xAD;
+            _queued_data[2] = 0xBE;
+            _queued_data[3] = 0xEF;
+
+            _queued_data[4] = 0x01; // command code
+            _queued_data[5] = 0x0A; // length
+            
+            // alpha
+            double_to_4bytes_little_endian(alpha, four_bytes);
+            _queued_data[6] = four_bytes[0];
+            _queued_data[7] = four_bytes[1];
+            _queued_data[8] = four_bytes[2];
+            _queued_data[9] = four_bytes[3];
+
+            // beta
+            double_to_4bytes_little_endian(beta, four_bytes);
+            _queued_data[10] = four_bytes[0];
+            _queued_data[11] = four_bytes[1];
+            _queued_data[12] = four_bytes[2];
+            _queued_data[13] = four_bytes[3];
+
+            // error code
+            _queued_data[14] = 1; // error
+            if (valid) _queued_data[14] = 0; // valid
+
+            // checksum
+            _queued_data[15] = compute_checksum(_queued_data, 4, 11);
+        }
+    }
+
+    uint8_t Generic_fssHardwareModel::compute_checksum(std::vector<uint8_t>& in, int starting_byte, int number_of_bytes)
+    {
+        uint32_t sum = 0;
+        uint8_t checksum;
+        for (int i = starting_byte; i < starting_byte + number_of_bytes; i++) {
+            sum += in[i];
+        }
+        checksum = (uint8_t)(sum & 0x000000FF);
+        return checksum;
+    }
+
+    void Generic_fssHardwareModel::double_to_4bytes_little_endian(double in, uint8_t out[4])
```

Comment:
Same comment as before with HWLIB, but sim_common maybe could house all these standard functions for re-use

Reply:
Same response as before.